compiler/rustc_codegen_ssa/src/mir/place.rs

```diff
@@ -403,15 +403,7 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
     }
 
     pub fn project_deref<Bx: BuilderMethods<'a, 'tcx, Value = V>>(&self, bx: &mut Bx) -> Self {
```

Comment:
This seems like it was an optimisation to me. Added in https://github.com/rust-lang/rust/pull/83941. cc @wesleywiser do you remember why you created this method instead of the more straightforward `load_operand().deref()`?

Reply:
I'm struggling to remember but I think I just didn't realize you could do that. From a quick glance, it seems to fine to me to replace this with `load_operand().deref()`.

Reply:
In that case I would perhaps explore inlining this method into its only use-site.